libtenzir/include/tenzir/split_regex.hpp

```diff
@@ -0,0 +1,50 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2025 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "tenzir/chunk.hpp"
+#include "tenzir/generator.hpp"
+
+#include <boost/regex.hpp>
+
+#include <optional>
+#include <string_view>
+
+namespace tenzir {
+
+inline auto split_regex(std::string separator) {
+  return [separator = std::move(separator)](generator<chunk_ptr> input)
+           -> generator<std::optional<std::string_view>> {
+    auto sep = separator;
+    auto buffer = std::string{};
+    auto expr = boost::regex{sep, boost::regex_constants::no_except};
+    for (auto&& chunk : input) {
+      if (! chunk || chunk->size() == 0) {
+        co_yield std::nullopt;
+        continue;
+      }
+      buffer.append(reinterpret_cast<const char*>(chunk->data()), chunk->size());
+      const auto first = buffer.cbegin();
+      auto current = first;
+      boost::match_results<std::string::const_iterator> what;
+      while (boost::regex_search(current, buffer.cend(), what, expr)) {
+        auto event = std::string_view{current, what[0].first};
+        co_yield event;
+        current = what[0].second;
+      }
```

Comment:
There's a very subtle bug here. Since you're getting the input piecemeal, you might return different results depending on how the input is arriving.

For example, if your pattern is `-*`, and you get as chunks `foo--` and then `---bar`, then you're currently returning `"foo"`, `""`, and `"bar"`, when you really shouldn't be returning the empty string in the middle.

Reply:
Right, didn't consider that. Should be fixed now.